Merge Updated Camera Controls

matthewm101 opened the pull request:
The camera now orbits a focus point that can be defined. By default, the focus is located a few units above the center of the grassy platform. Zoom is also enabled, and the camera can be rotated using the arrow keys.

matthewm101 committed:
Updated camera controls

matthewm101 committed:
Camera controller tweaking

matthewm101 committed:
Merged camera controls

matthewm101 committed:
Merge branch 'Cinnamon18-master'

matthewm101 committed:
Camera Control Fix

Cinnamon18 commented:
ty for the pr! do you think you could switch arrow keys to pan? i think we talked with the team and thought that would be more intuitive

matthewm101 committed:
Resolved merge conflicts

matthewm101 committed:
Adding camera panning

matthewm101 commented:
The camera can now be panned with the arrow keys or WASD. The camera is only allowed to look at the floor tile where the chickens are, but the focus of the camera can move anywhere inside the pen.

matthewm101 committed:
Hens no longer fly into the void

Instead of turning by directly setting a hen's rotation, hens now receive an angular impulse. This makes the clipping no longer occur.

matthewm101 commented:
The hen clipping fix was added.
